packages/shared/lib/contexts/governance/actions/registerProposalsFromPrimaryNode.ts

```diff
@@ -0,0 +1,34 @@
+import { get } from 'svelte/store'
+
+import { selectedAccount } from '@core/account/stores'
+import { activeProfile } from '@core/profile/stores'
+
+import { getVotingEventIds, isProposalAlreadyAddedForSelectedAccount } from '../utils'
+
+import { registerParticipationEvent } from './registerParticipationEvent'
+
+export async function registerProposalsFromPrimaryNode(): Promise<void> {
```

Comment:
Can this be more generic by just passing in a node and then we can decide what nodes to use where we call this. i.e. we don't usually have a primary node